### PR TITLE
378 expose authorize user endpoint

### PR DESCRIPTION
**Pull request by ivanzaqqa**
#378 

### Breakdown:
- expose authorize user endpoint

### CODE REVIEW

**Comment by ivanzaqqa** [comment]
if any wrong please tell me mr. andy, thanks

File: app/controllers/user_controller.py [hunk 1]
```diff
@@ -89,3 +90,16 @@ def redeemcode(request, user):
             return BaseController.send_response_api(result['data'], result['message'])
         else:
             return BaseController.send_error_api(None, 'invalid payload')
+
+    @staticmethod
+    def password_require(request):
+        username = request.json['username'] if 'username' in request.json else None
```
Comment (andy-shi88) [inline app/controllers/user_controller.py]: I'd rather not needing to provide username, let's add token_required in the route and fetch username from the token/

File: app/controllers/user_controller.py [hunk 2]
```diff
@@ -89,3 +90,16 @@ def redeemcode(request, user):
...
+        password = request.json['password'] if 'password' in request.json else None
+        if username and password:
+            admin = userservice.get_user(username)
```
Comment (andy-shi88) [inline app/controllers/user_controller.py]: naming convention: I don't think it has anything to do with admin right? try give descriptive variable name.

**Comment by ivanzaqqa** [comment]
thanks too for your guidance mas

**Pull request merged**